js/modules/k6/grpc/client.go

```diff
@@ -212,15 +206,73 @@ func (t transportCreds) ClientHandshake(ctx context.Context,
 	return out, auth, err
 }
 
+func (c *Client) dial(
+	ctx context.Context,
+	addr string,
+	reflect bool,
+	state *lib.State,
+	isPlaintext bool,
+	errc chan error,
+) {
+	opts := []grpc.DialOption{
+		grpc.WithBlock(),
+		grpc.FailOnNonTempDialError(true),
+		grpc.WithStatsHandler(c),
+	}
+
+	if ua := state.Options.UserAgent; ua.Valid {
+		opts = append(opts, grpc.WithUserAgent(ua.ValueOrZero()))
+	}
+
+	if !isPlaintext {
+		tlsCfg := state.TLSConfig.Clone()
+		tlsCfg.NextProtos = []string{"h2"}
+
+		// TODO(rogchap): Would be good to add support for custom RootCAs (self signed)
+
+		// (rogchap) we create a wrapper for transport credentials so that we can report
+		// on any TLS errors.
+		creds := transportCreds{
+			credentials.NewTLS(tlsCfg),
+			errc,
+		}
+		opts = append(opts, grpc.WithTransportCredentials(creds))
+	}
+
+	if isPlaintext {
+		opts = append(opts, grpc.WithInsecure())
+	}
+
+	dialer := func(ctx context.Context, addr string) (net.Conn, error) {
+		return state.Dialer.DialContext(ctx, "tcp", addr)
+	}
+	opts = append(opts, grpc.WithContextDialer(dialer))
+
+	var err error
+	c.conn, err = grpc.DialContext(ctx, addr, opts...)
+	if err != nil {
+		errc <- err
+		return
+	}
+	if reflect {
+		err := c.reflect(ctx)
+		if err != nil {
+			errc <- err
+			return
+		}
+	}
+
+	errc <- nil
```

Comment:
Do you think we can reduce nesting level and make code simpler? E.g.:

```suggestion
	if !reflect {
		errc <- nil

		return
	}

	errc <- c.reflect(ctx)
```